src/deepsparse/pipeline.py

```diff
@@ -176,22 +174,113 @@ def __init__(
         self.onnx_file_path = self.setup_onnx_file_path()
         self.engine = self._initialize_engine()
 
-    def __call__(self, *args, **kwargs) -> Union[BaseModel, Future]:
-        _default_key_val = ("_DEFAULT",)
-        executor = kwargs.get("executor", _default_key_val)
+        self._batch_size = self._batch_size or 1
 
-        if executor is _default_key_val:  # do not use ==
-            # use executor created during initialization
-            executor = self.executor
+    def __call__(self, *args, **kwargs) -> BaseModel:
+        if "engine_inputs" in kwargs:
+            raise ValueError(
+                "invalid kwarg engine_inputs. engine inputs determined "
+                f"by {self.__class__.__qualname__}.parse_inputs"
+            )
+
+        # parse inputs into input_schema
+        pipeline_inputs = self.parse_inputs(*args, **kwargs)
+        if not isinstance(pipeline_inputs, self.input_schema):
+            raise RuntimeError(
+                f"Unable to parse {self.__class__} inputs into a "
+                f"{self.input_schema} object. Inputs parsed to {type(pipeline_inputs)}"
+            )
+
+        # batch size of the inputs may be `> self._batch_size` at this point
+        engine_inputs: List[numpy.ndarray] = self.process_inputs(pipeline_inputs)
+        if isinstance(engine_inputs, tuple):
+            engine_inputs, postprocess_kwargs = engine_inputs
         else:
-            # use passed in executor
-            executor = kwargs.pop("executor")
+            postprocess_kwargs = {}
+
+        # split inputs into batches of size `self._batch_size`
+        batches = self.split_engine_inputs(engine_inputs, self._batch_size)
```

Comment:
adding as default flow even for bs1 real time looks fine, just to be sure, let's take a quick benchmark of overhead added by doing the split/submit/join for bs1

Reply:
Yep I had done a benchmark of this - no overhead added